src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+        } else {
+            queryFilterValidation();
+        }
+    }
+
+    private long[] startEndTimeRangeWithIntervals(int numOfIntervals) {
+        long delayMillis = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getWindowDelay())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
+        long dataEndTime = Instant.now().toEpochMilli() - delayMillis;
+        long detectorInterval = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getDetectionInterval())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
+        long dataStartTime = dataEndTime - ((long) (numOfIntervals) * detectorInterval);
+        return new long[] { dataStartTime, dataEndTime };
+    }
+
+    private void queryFilterValidation() {
+        long[] startEnd = startEndTimeRangeWithIntervals(NUM_OF_INTERVALS_CHECKED_FILTER);
+        long dataEndTime = startEnd[1];
+        long dataStartTime = startEnd[0];
+        RangeQueryBuilder rangeQuery = new RangeQueryBuilder(anomalyDetector.getTimeField())
+            .from(dataStartTime)
+            .to(dataEndTime)
+            .format("epoch_millis");
+        BoolQueryBuilder internalFilterQuery = QueryBuilders.boolQuery().must(rangeQuery).must(anomalyDetector.getFilterQuery());
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+            .query(internalFilterQuery)
+            .size(1)
+            .terminateAfter(1)
+            .timeout(requestTimeout);
+        SearchRequest searchRequest = new SearchRequest(anomalyDetector.getIndices().get(0)).source(searchSourceBuilder);
+        client
+            .search(
+                searchRequest,
+                ActionListener.wrap(searchResponse -> onQueryFilterSearch(searchResponse), exception -> { onFailure(exception); })
+            );
+    }
+
+    private void onQueryFilterSearch(SearchResponse response) throws IOException {
+        if (response.getHits().getTotalHits().value <= 0) {
+            List<String> filterError = new ArrayList<>();
+            filterError
+                .add(
+                    "query filter is potentially wrong as no hits were found at all or no historical data in last "
+                        + NUM_OF_INTERVALS_CHECKED_FILTER
+                        + " intervals"
+                );
+            suggestedChangesMap.put("filter_query", filterError);
+            validateAnomalyDetectorResponse();
+        } else {
+            long timestamp = (long) response.getHits().getHits()[0].getSourceAsMap().get("timestamp");
+            featureQueryValidation(timestamp);
+        }
+    }
+
+    private void featureQueryValidation(long startTimeStamp) throws IOException {
+        long delayMillis = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getWindowDelay())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
```

Comment:
You may need to have a util method to convert IntervalTimeConfiguration duration to Millisec. Looks like it can be re-used multiple times

Reply:
fixed